contracts/Rln.sol

```diff
@@ -86,7 +88,8 @@ contract RLN {
 		members[_pubkeyIndex] = 0;
 
 		// refund deposit
-		(bool sent, _) = receiver.call{value: MEMBERSHIP_DEPOSIT}("");
+		uint256 withdrawableDeposit = MEMBERSHIP_DEPOSIT * (WAD - PROTOCOL_FEES_PERCENTAGE) / WAD
```

Comment:
To dave on the gas, isn't it better to do this calculation once (e.g., in the constructor or when defining the member fields) and save the result in a variable like 'withdrawable_amount`  and then reuse the calculated value in this function call?

Reply:
Great! Thanks for the insights, the result is so unexpected! 
Considering this observation, do you think it is more cost-efficient to keep the `PROTOCOL_FEES_PERCENTAGE` as a hardcoded value or to pass it as an argument to the constructor in which case it becomes an immutable variable (hence it causes a higher execution cost when it comes to the `withdraw` function. 
I already spoke with @kgrgpg and he is going to inspect ENS contract to see how the service fee is modeled there.  NIn the meanwhile, would be also interested to know your inputs @richard-ramos

Reply:
I'm not an expert in this topic, but for costs, I believe it should be more efficient to have the percentage hardcoded, but it has a disadvantage of not being able to change this value later unless whoever uses the contract also implements some mechanism like making the contract upgradeable which could allow the contract owner to change the implementation.

Reply:
Just noticed that if the `PROTOCOL_FEES_PERCENTAGE` is set in the constructor, it does not have to be `mutable`, hence making it a parameter of the constructor would not cause any additional gas cost in the `withdraw` function.